saleor/webhook/observability_reporter/payloads.py

```diff
@@ -0,0 +1,318 @@
+
+
+class ObservabilityEventPayload(TypedDict):
+    event_type: ObservabilityEventTypes
+
+
+class GraphQLOperationPayload(TypedDict):
+    name: Optional[JsonTruncText]
+    operation_type: Optional[str]
+    query: Optional[JsonTruncText]
+    result: Optional[JsonTruncText]
+
+
+class RequestPayload(TypedDict):
+    id: str
+    method: str
+    url: str
+    time: float
+    headers: Dict[str, str]
+    content_length: int
+
+
+class ResponsePayload(TypedDict):
+    headers: Dict[str, str]
+    status_code: Optional[int]
+    content_length: int
+
+
+class AppPayload(TypedDict):
+    id: str
+    name: str
+
+
+class GraphQLOperationsPayload(TypedDict):
+    count: int
+    operations: List[GraphQLOperationPayload]
+
+
+class ApiCallPayload(ObservabilityEventPayload):
+    request: RequestPayload
+    response: ResponsePayload
+    app: Optional[AppPayload]
+    gql_operations: GraphQLOperationsPayload
+
+
+class EventDeliveryDataPayload(TypedDict):
+    content_length: int
+    body: JsonTruncText
+
+
+class EventDeliveryPayload(TypedDict):
+    id: str
+    status: str
+    event_type: str
+    payload: EventDeliveryDataPayload
+
+
+class WebhookPayload(TypedDict):
+    id: str
+    name: str
+    target_url: str
+    subscription_query: Optional[JsonTruncText]
+
+
+class AttemptPayload(TypedDict):
+    id: str
+    time: datetime
+    duration: Optional[float]
+    status: str
+    next_retry: Optional[datetime]
+
+
+class RequestHeadersPayload(TypedDict):
+    headers: Dict[str, str]
+
+
+class ResponseWithBodyPayload(ResponsePayload):
+    body: JsonTruncText
+
+
+class EventDeliveryAttemptPayload(ObservabilityEventPayload):
+    event_delivery_attempt: AttemptPayload
+    request: RequestHeadersPayload
+    response: ResponseWithBodyPayload
+    event_delivery: EventDeliveryPayload
+    webhook: WebhookPayload
+    app: AppPayload
+
+
+def _json_serialize(obj: Any, pretty=False):
+    if pretty:
+        return json.dumps(obj, indent=2, ensure_ascii=True, cls=CustomJsonEncoder)
+    return json.dumps(obj, ensure_ascii=True, cls=CustomJsonEncoder)
+
+
+TRUNC_PLACEHOLDER = JsonTruncText(truncated=False)
+EMPTY_TRUNC = JsonTruncText(truncated=True)
+
+
+GQL_OPERATION_PLACEHOLDER = GraphQLOperationPayload(
+    name=TRUNC_PLACEHOLDER,
+    operation_type="subscription",
+    query=TRUNC_PLACEHOLDER,
+    result=TRUNC_PLACEHOLDER,
+)
+GQL_OPERATION_PLACEHOLDER_SIZE = len(_json_serialize(GQL_OPERATION_PLACEHOLDER))
+
+
+def serialize_gql_operation_result(
+    operation: "GraphQLOperationResponse", bytes_limit: int
+) -> Tuple[GraphQLOperationPayload, int]:
+    bytes_limit -= GQL_OPERATION_PLACEHOLDER_SIZE
+    if bytes_limit < 0:
+        raise ValueError()
+    anonymize_gql_operation_response(operation, SENSITIVE_GQL_FIELDS)
+    name: Optional[JsonTruncText] = None
+    operation_type: Optional[str] = None
+    query: Optional[JsonTruncText] = None
+    result: Optional[JsonTruncText] = None
+    if operation.name:
```

Comment:
Not sure if this is already added, but wouldn't be good to include a some const string like or something that would give you info, that te response that you see was limited.

Reply:
Yes, that's already added if I understand the question correctly. `JsonTruncText` object serializes to `text: str, truncated: bool`